services/document_service.go

```diff
@@ -1,18 +1,37 @@
 package services
 
 import (
-	"time"
-
 	"github.com/docentre/docentre/models"
 	"github.com/docentre/docentre/repositories"
 )
 
 func CreateDocument(authorID uint) (models.Document, error) {
-	doc := models.Document{AuthorID: authorID, LastEditDate: time.Now()}
-	result := repositories.DB.Create(&doc)
+	doc := models.Document{AuthorID: authorID}
+	result := repositories.DB.Omit("ApprovedDate, ApproverID, ViewerID").Create(&doc)
 	if result.Error != nil {
 		return models.Document{}, result.Error
 	}
 
 	return doc, nil
 }
+
+func UpdateDocument(documentID uint, authorID uint, title string, content string, appendix string, status string, approverID uint) (uint, error) {
+
+	if approverID == 0 {
```

Comment:
`0` is a magic number; consider introducing a explaining variable:
```suggestion
    noApproverYet := 0;
	if approverID == noApproverYet {
```